src/sidebar/store/modules/selection.js

```diff
@@ -257,63 +280,57 @@ const update = {
 
 const actions = util.actionTypes(update);
 
-function select(annotations) {
+/**
+ * Set the currently selected annotation IDs. This will replace the current
+ * selection. All provided annotation ids will be set to `true` in the selection.
+ *
+ * @param {string[]} ids - Idenfiers of annotations to select
+ */
+function selectAnnotations(ids) {
+  const selection = {};
+  ids.forEach(id => (selection[id] = true));
   return {
     type: actions.SELECT_ANNOTATIONS,
-    selection: freeze(annotations),
+    selection,
   };
 }
 
 /**
- * Set the currently selected annotation IDs.
+ * Toggle the selected state for the annotations in `toggledAnnotations`:
+ * unselect any that are selected; select any that are unselected.
+ *
+ * @param {string[]} toggleIds - identifiers of annotations to toggle
  */
-function selectAnnotations(ids) {
-  return select(toSet(ids));
-}
-
-/** Toggle whether annotations are selected or not. */
-function toggleSelectedAnnotations(ids) {
-  return function (dispatch, getState) {
-    const selection = Object.assign(
-      {},
-      getState().selection.selectedAnnotationMap
-    );
-    for (let i = 0; i < ids.length; i++) {
-      const id = ids[i];
-      if (selection[id]) {
-        delete selection[id];
-      } else {
-        selection[id] = true;
-      }
-    }
-    dispatch(select(selection));
+function toggleSelectedAnnotations(toggleIds) {
```

Comment:
Eek! - The previous code here was pretty complex for what it did.

Reply:
I know, rite?!